cert.go

```diff
@@ -51,6 +51,13 @@ func (m *mkcert) makeCert(hosts []string) {
 	serialNumber, err := rand.Int(rand.Reader, serialNumberLimit)
 	fatalIfErr(err, "failed to generate serial number")
 
+	var extKeyUsage []x509.ExtKeyUsage
+	if m.client {
+		extKeyUsage = []x509.ExtKeyUsage{x509.ExtKeyUsageClientAuth}
```

Comment:
Why not just add ClientAuth as a usage if the flag is set?

Reply:
In my case, I wanted separate certificates. If the certificate is meant for client auth, then it doesn't need the server auth extension. To me, it would seem weird that a client certificate could be used for server authentication as well.